app.js

```diff
@@ -1,46 +1,138 @@
 "use strict";
 
-const express = require("express"); // web application framework
+const express = require("express");
+const bodyParser = require("body-parser");
 const app = express();
 const child_process = require("child_process");
+const fs = require("fs");
+const pngToJpeg = require("png-to-jpeg");
 
-// set server to use preconfigured port or 8000
-app.set("port", (process.env.PORT || 8000));
+app.set("port", (process.env.PORT || 8000)); // set server to use preconfigured port or 8000
+app.use(bodyParser.json({limit: "50mb"}));
+app.use(bodyParser.urlencoded({limit: "50mb", extended: true, parameterLimit: 50000}));
+
+/**
+ * Writes a base64 string for a JPEG file to disk as a JPEG file
+ * @param {string} base64String - base64 string of JPEG file
+ */
+function writeJpegToDisk(base64String) {
+    let base64Data = base64String.replace(/^data:image\/jpeg;base64,/, ""); // stripping for decoding
+    // writing file to disk as "out.jpeg"
+    fs.writeFile("out.jpeg", base64Data, "base64", function (error) {
+        if (error !== null) {
+            console.log(error);
+        }
+    });
+}
+
+/**
+ * Writes a base64 string for a PNG file to disk as a JPEG file
+ * @param {string} base64String - base64 string of PNG file
+ */
+function writePngToDisk(base64String) {
+    const buffer = new Buffer(base64String.split(/,\s*/)[1], "base64");
+    // using middleware to change image from PNG to JPEG
+    pngToJpeg({quality: 90})(buffer).then(function (output) {
+        // writing file to disk as "out.jpeg"
+        fs.writeFile("out.jpeg", output, function (error) {
+            if (error !== null) {
+                console.log(error);
+            }
+        });
+    });
+}
+
+/**
+ * Confirms the file type of a base64 string, whether PNG, JPEG, or other
+ * @param {string} base64String - base64 string of file
+ * @returns {string} fileType - assignment of file type, which will either be PNG, JPEG, or undefined
+ */
+function checkFileType(base64String) {
+    let fileType = base64String.substring(0, 15); // getting the data/image type label
+    if (fileType.includes("png")) {
```

Comment:
This throws an exception if fileType is undefined. This function should be self-contained for the most part. If the logic is copied anywhere, someone won't think that this logic isn't enforcing non-null parameters